src/zap.ts

```diff
@@ -47,18 +47,18 @@ export function getSwapAmountDepositedFromZapping(
   pactFeeBps: bigint,
 ): bigint {
   const poolFee = feeBps - pactFeeBps;
-  const a = (-1n * (FEE_PRECISION + pactFeeBps)) / FEE_PRECISION;
+  const a = pactFeeBps - FEE_PRECISION;
   const b =
     (-2n * totalAmount * FEE_PRECISION +
       zapAmount * poolFee +
       totalAmount * feeBps) /
     FEE_PRECISION;
   const c = totalAmount * zapAmount;
 
-  const delta = b * b - 4n * a * c;
+  const delta = b * b - (4n * a * c) / FEE_PRECISION;
   let result;
   if (b < 0) {
-    result = (-1n * b - isqrt(delta)) / (2n * a);
+    result = (-1n * b - isqrt(delta)) / ((2n * a) / FEE_PRECISION);
```

Comment:
we might acquire better precision here `(-1n * b - isqrt(delta)) / ((2n * a) / FEE_PRECISION);` -> `(-1n * b - isqrt(delta)) * FEE_PRECISION / (2n * a)`. If I remember correctly bigints can be arbitrarily long, so we can introduce this small change